### DISCUSSION TITLE
D3DMetal piggybacking | unexpected interoperability

### DISCUSSION THREAD

**Discussion by Rastafabisch** [3Shain/dxmt#18] [Show and tell]
I was playing a little bit around a realised a surprising (at least for me?) behavior. I could get the d3d11 and dig DLLs to load (without wine metal.dll or any other necessary library/API provider) on fairly old wine builds (tested 7.??+) whenever D3DMetal was installed. Additionally this also did enable functionality, that standalone DXMT on a fully customized 9.12 build did not provide. I did ensure, stat said DXMT library did load while apples D3DMetal enabled DLLs did not. I somewhat puzzled.

A coincidental preliminary test did also confirm that DXMTs libraries have been used (Stock D3DMetal could enable the MetalHUD, while I just couldn't get it to work with this Frankenstein conglomerate of libraries.) 

Just wanted to report this because I found this to be quite fascinating. 

| Test scenario | The Witcher 3 |
|---|---|
|9.12 DXMT|launching with black screen| 
|CrossOver wine 22.1.1 (with D3DMetal [1.1/2.0b2/2.0b3])|running fine|
|CrossOver wine 22.1.1 (no D3DMetal + DXMT d3d11 & dxgi)|complaing about winemetal.dll|
|CrossOver wine 22.1.1 (with D3DMetal [1.1/2.0b2/2.0b3] + DXMT d3d11 & dxgi)|running fine as well|

[Wine-Log (D3DMetal+DXMT)](https://github.com/user-attachments/files/17171649/D3DMetal%2BDXMT.zip)

**Comment by 3Shain** [comment]
I don't think DXMT is working for you at all. IIRC CrossOver wine 22.1.1 is based on Wine 7.7. However, DXMT only supports Wine 8+ with no exception. Also from your log I haven't found any output from DXMT, but I'm pretty sure D3DMetal is taking effect.

I haven't tested The Witcher 3. It's possible that some features are still missing. If you can make sure DXMT is loaded for Wine 9.12 (I'm not sure where you get it from, because DXMT requires a patch to Wine to work, that I don't think I have publicly documented), it would be more helpful to provide the log from DXMT.
